src/vse_sync_pp/analyzers/pmc.py

```diff
@@ -0,0 +1,134 @@
+    for clock_class in clock_class_count.values():
+        clock_class["transitions"] = {STATE_NAMES[k]:v for (k,v) in clock_class["transitions"].items()}
+    return named_clock_class_count
+
+class ClockStateAnalyzer(Analyzer):
+    """Analyze clock state
+    """
+    id_ = 'phc/gm-settings'
+    parser = id_
+    def __init__(self, config):
+        super().__init__(config)
+        # minimum test duration for a valid test
+        self._duration = config.parameter('min-test-duration/s')
+        self.state = None
+        self.transition_count = 0
+        self.clock_class_count = {
+            STATE_FREERUN: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_LOCKED: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_HOLDOVER_IN_SPEC: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_HOLDOVER_OUT_OF_SPEC1: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_HOLDOVER_OUT_OF_SPEC2: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_HOLDOVER_OUT_OF_SPEC3: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+        }
+    def prepare(self, rows):
+        idx = 0
+        try:
+            tstart = rows[0].timestamp
+        except IndexError:
+            pass
+        else:
+            while idx < len(rows):
+                if tstart <= rows[idx].timestamp:
+                    break
+                idx += 1
+        return super().prepare(rows[idx:])
+    def test(self, data):
+        if len(data) == 0:
+            return (False, "no data")
+        
+        for index, row in data.iterrows():
+            clock_class = row['clock_class']
+ 
+            if (self.state is None) and (clock_class in STATE_TRANSITION):
+                self.state = clock_class
+            else:
+                if clock_class != self.state:
+                    self.transition_count += 1
+                
+                if clock_class not in STATE_TRANSITION:
+                    return (False, f"wrong clock class {clock_class}")
+                
+                if is_transition_allowed(self.state, clock_class):
+                    self.clock_class_count[self.state]["transitions"][clock_class] += 1
+                    self.state = clock_class
+                    self.clock_class_count[clock_class]["count"] += 1
+                else:
+                    return (False, "wrong state transition")
+
+        if data.iloc[-1].timestamp - data.iloc[0].timestamp < self._duration:
+            return (False, "short test duration")
+        if len(data) - 1 < self._duration:
+            return (False, "short test samples")
+        return (True, None)
+    def explain(self, data):
+        if len(data) == 0:
+            return {}
+
+        return {
+            'duration': data.iloc[-1].timestamp - data.iloc[0].timestamp,
+            'clock_class_count': get_named_clock_class_result(self.clock_class_count),
+            'total_transitions': self.transition_count,
+        }
```

Comment:
We should review together how we want this data to appear in the test report. (How we render this currently has restrictions.)